### PR TITLE
Ensure generated DOT file doesn't change unless state machine changes

### PR DESCRIPTION
**Pull request by cezarsa**
By using slices and sorting them by state and trigger values we ensure that ToGraph() will always return the same output for the same state machine.

This is important in situations where ToGraph() is used to generate a file that's committed to a repository to ensure it doesn't
change on every commit.

### CODE REVIEW

File: graph.go [hunk 1]
```diff
@@ -15,12 +16,20 @@ func (g *graph) FormatStateMachine(sm *StateMachine) string {
 	var sb strings.Builder
 	sb.WriteString("digraph {\n\tcompound=true;\n\tnode [shape=Mrecord];\n\trankdir=\"LR\";\n\n")
 
-	for _, sr := range sm.stateConfig {
+	var stateList []*stateRepresentation
```
Comment (qmuntal) [inline graph.go:19]: Preallocate the slice with the size of sm.stateConfig

**Review by qmuntal — CHANGES_REQUESTED** [review]

**Comment by qmuntal** [comment]
Thanks for this contribution

File: graph.go (hunk 1 above)
Comment (cezarsa) [reply graph.go]: Done. Did the same for `sr.TriggerBehaviours` but there it will still grow if there's more than one behaviour for the same trigger.

**Review by qmuntal — APPROVED** [review]
LGTM!

**Pull request merged**